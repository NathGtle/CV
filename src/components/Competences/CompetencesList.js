import React from 'react'
import Competence from './Competence.js'

const CompetencesList = ({ competenceslist }) => (
  <>
    <div className="headlineCompetence">{competenceslist.label}</div>
    {competenceslist.values.map((competence, index) => {
      return <Competence key={index} competence={competence} />
    })}
  </>
)

export default CompetencesList
