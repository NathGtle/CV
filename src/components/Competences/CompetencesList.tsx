import React from 'react'
import PropTypes from 'prop-types'
import Competence from './Competence'

interface CompetenceslistInterface {
  competencesList: {
    label: string
    values: Array<{ value: string; name: string }>
  }
}

const CompetencesList = ({ competencesList }: CompetenceslistInterface) => {
  if (!competencesList) return null
  return (
    <React.Fragment>
      <div className="headline headline_alt">{competencesList.label}</div>
      {competencesList.values
        .sort((a, b) => parseFloat(b.value) - parseFloat(a.value))
        .map((competence, index) => {
          return <Competence key={index} competence={competence} />
        })}
    </React.Fragment>
  )
}

CompetencesList.propTypes = {
  competencesList: PropTypes.shape({
    label: PropTypes.string.isRequired,
    values: PropTypes.array.isRequired,
  }),
}

export default CompetencesList
