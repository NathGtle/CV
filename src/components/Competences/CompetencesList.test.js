import React from 'react'
import renderer from 'react-test-renderer'
import CompetencesList from './CompetencesList'

const mocks = {
  competencesList: {
    label: 'Programmation',
    values: [
      {
        name: 'Javascript',
        value: '4.5',
      },
      {
        name: 'Node.JS',
        value: '3.6',
      },
    ],
  },
}

jest.mock('./Competence', () => 'mock-competence')

describe('CompetencesList component', () => {
  it('should render correctly with props', () => {
    const component = renderer.create(<CompetencesList {...mocks} />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly without props', () => {
    const component = renderer.create(<CompetencesList />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
