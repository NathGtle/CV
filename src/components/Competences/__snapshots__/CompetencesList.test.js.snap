// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompetencesList component should render correctly with props 1`] = `
Array [
  <div
    className="headline headline_alt"
  >
    Programmation
  </div>,
  <mock-competence
    competence={
      Object {
        "name": "Javascript",
        "value": "4.5",
      }
    }
  />,
  <mock-competence
    competence={
      Object {
        "name": "Node.JS",
        "value": "3.6",
      }
    }
  />,
]
`;

exports[`CompetencesList component should render correctly without props 1`] = `null`;
